mailgun/route_handlers.py

```diff
@@ -39,8 +41,22 @@ def handle_mailing_list_email_route(request):
 
     member_addresses = [m['address'] for m in ml.members]
     if ml.access_level == MailingList.ACCESS_LEVEL_MEMBERS and sender not in member_addresses:
-        logger.info("Sending mailing list bounce back email to %s for mailing list %s", sender, recipient)
-        email_template = get_template('mailgun/non_member_bounce_back.html')
+        logger.info(
+            "Sending mailing list bounce back email to %s for mailing list %s because the sender was not a member",
+            sender,
+            recipient
+        )
+        email_template = get_template('mailgun/email/bounce_back_non_member.html')
+        content = email_template.render(Context({'mailing_list_address': recipient}))
+        subject = "Undeliverable mail sent to %s" % recipient
+        ml.send_mail(sender, subject, html=content)
```

Comment:
Minor : Since the 'content = email_template.render..' block is the same and occurs  for both the conditions( in lines 50 and 60), this could  be pulled out side the if -else block and placed outside.